File: python-pr/52/python/code-quality-readability-maintainability/file_processor_06.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def process_file(f):
    try:
        with open(f) as file:
            lines = file.readlines()
            processed = []
            for line in lines:
                if line.strip():
                    words = line.split()
                    filtered_words = [w for w in words if len(w) > 3]
                    if filtered_words:
                        processed.append(' '.join(filtered_words))
            return processed
    except:
        return []
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/file_processor_06.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function processFile(f) {
    const fs = require('fs');
    try {
        let data = fs.readFileSync(f, 'utf8');
        let lines = data.split('\n');
        let processed = [];
        for (let line of lines) {
            if (line.trim()) {
                let words = line.split(' ');
                let filteredWords = words.filter(w => w.length > 3);
                if (filteredWords.length > 0) {
                    processed.push(filteredWords.join(' '));
                }
            }
        }
        return processed;
    } catch (e) {
        return [];
    }
}
// {/fact}